3.2.2 release  add task-pure delete some task module to reduce the size of the deployment package

wangxj3 opened the pull request:
<!--Thanks very much for contributing to Apache DolphinScheduler, we are happy that you want to help us improve DolphinScheduler! -->

## Purpose of the pull request
 add task pure remove the plugin of 
dolphinscheduler-task-sagemaker
dolphinscheduler-task-dvc
dolphinscheduler-task-openmldb
dolphinscheduler-task-mlflow
dolphinscheduler-task-zeppelin
dolphinscheduler-task-sqoop
dolphinscheduler-task-chunjun
dolphinscheduler-task-dms
dolphinscheduler-task-datafactory
dolphinscheduler-task-dinky
dolphinscheduler-task-pytorch
dolphinscheduler-task-datasync
dolphinscheduler-task-kubeflow
dolphinscheduler-task-emr
dolphinscheduler-task-pigeon
dolphinscheduler-task-flink-stream；

add datasource pure remove the plugin of 
dolphinscheduler-datasource-athena
dolphinscheduler-datasource-azure-sql
dolphinscheduler-datasource-dameng
dolphinscheduler-datasource-databend
dolphinscheduler-datasource-db2
dolphinscheduler-datasource-hana
dolphinscheduler-datasource-kyuubi
dolphinscheduler-datasource-oceanbase
dolphinscheduler-datasource-presto
dolphinscheduler-datasource-redshift
dolphinscheduler-datasource-sagemaker
dolphinscheduler-datasource-snowflake
dolphinscheduler-datasource-trino
dolphinscheduler-datasource-vertica
dolphinscheduler-datasource-zeppelin
dolphinscheduler-datasource-clickhouse
dolphinscheduler-datasource-starrocks
dolphinscheduler-datasource-sqlserver
dolphinscheduler-datasource-oracle；

add storage pure remove the plugin of 
dolphinscheduler-storage-obs
dolphinscheduler-storage-abs
dolphinscheduler-storage-gcs
dolphinscheduler-storage-oss

add registry pure remove the plugin of 
dolphinscheduler-registry-jdbc
dolphinscheduler-registry-etcd

## Brief change log

<!--*(for example:)*
- *Add maven-checkstyle-plugin to root pom.xml*
-->

## Verify this pull request

<!--*(Please pick either of the following options)*-->

This pull request is code cleanup without any test coverage.

*(or)*

This pull request is already covered by existing tests, such as *(please describe tests)*.

(or)

This change added tests and can be verified as follows:

<!--*(example:)*
- *Added dolphinscheduler-dao tests for end-to-end.*
- *Added CronUtilsTest to verify the change.*
- *Manually verified the change by testing locally.* -->

(or)

If your pull request contain incompatible change, you should also add it to `docs/docs/en/guide/upgrede/incompatible.md`

Commit:
add version of 3.2.2

wangxj3 committed:
Update Chart.yaml Rollback APPversion

Commit:
add task-pure delete some task module to reduce the size of the deployment package

ruanwenjun reviewed (changes requested):
Why submit this PR to dev?

Commit:
remove task pure plugin

Commit:
fix ci

Commit:
add pure datasource plugin

Commit:
remove ck，sr from datasource pure

Commit:
remove task plugin chunjun ，dms ，datafactory from pure

Commit:
add api pom dependency

Commit:
spotless:apply

Commit:
add dependency license

Commit:
[maven-release-plugin] prepare release 3.2.2

Commit:
[maven-release-plugin] prepare for next development iteration

Commit:
remove datasource-pure：
dolphinscheduler-datasource-sqlserver
dolphinscheduler-datasource-oracle

remove task-pure：
dolphinscheduler-task-dinky
dolphinscheduler-task-pytorch
dolphinscheduler-task-datasync
dolphinscheduler-task-kubeflow
dolphinscheduler-task-emr
dolphinscheduler-task-pigeon
dolphinscheduler-task-flink-stream

Commit:
Merge branch '3.2.2-release' of github.com:wangxj3/dolphinscheduler into 3.2.2-release

Commit:
add register pure remove jdbc and etcd

Commit:
add dolphinscheduler-storage-pure remove task of
dolphinscheduler-storage-obs
dolphinscheduler-storage-abs
dolphinscheduler-storage-gcs
dolphinscheduler-storage-oss

Commit:
mvn spotless:apply
